plugins/nf-validation/src/main/nextflow/validation/CustomEvaluators/SchemaEvaluator.groovy

```diff
@@ -0,0 +1,67 @@
+package nextflow.validation
+
+import dev.harrel.jsonschema.Evaluator
+import dev.harrel.jsonschema.EvaluationContext
+import dev.harrel.jsonschema.JsonNode
+import nextflow.Nextflow
+import nextflow.Global
+import groovy.json.JsonGenerator
+import org.json.JSONArray
+
+import groovy.util.logging.Slf4j
+import java.nio.file.Path
+import java.nio.file.Files
+
+@Slf4j
+class SchemaEvaluator implements Evaluator {
+    // Evaluate the file using the given schema
+
+    private final String schema
+
+    SchemaEvaluator(String schema) {
+        this.schema = schema
+    }
+
+    @Override
+    public Evaluator.Result evaluate(EvaluationContext ctx, JsonNode node) {
+        // To stay consistent with other keywords, types not applicable to this keyword should succeed
+        if (!node.isString()) {
+            return Evaluator.Result.success()
+        }
+
+        def String value = node.asString()
+
+        // Actual validation logic
+        def Path file = Nextflow.file(value)
+        // Don't validate if the file does not exist or is a directory
+        if(!file.exists() && !file.isDirectory()) {
```

Comment:
Should this not be `if(!file.exists() || file.isDirectory())`?

Reply:
You are right! Thanks for pointing out :100: